batch/batch/front_end/front_end.py

```diff
@@ -672,6 +719,68 @@ async def get_batches_v2(request, userdata):  # pylint: disable=unused-argument
     return json_response({'batches': batches})
 
 
+async def _query_job_groups(request, batch_id: int, job_group_id: int, last_child_job_group_id: Optional[int]):
+    db: Database = request.app['db']
+
+    record = await db.select_and_fetchone(
+        """
+SELECT 1
+FROM job_groups
+LEFT JOIN batches ON batches.id = job_groups.batch_id
+LEFT JOIN batch_updates
+  ON job_groups.batch_id = batch_updates.batch_id AND job_groups.update_id = batch_updates.update_id
+WHERE job_groups.batch_id = %s AND job_groups.job_group_id = %s AND NOT deleted AND (batch_updates.committed OR job_groups.job_group_id = %s);
+""",
+        (batch_id, job_group_id, ROOT_JOB_GROUP_ID),
+    )
+    if not record:
+        raise NonExistentJobGroupError(batch_id, job_group_id)
+
+    sql, sql_args = parse_list_job_groups_query_v1(batch_id, job_group_id, last_child_job_group_id)
+    job_groups = [job_group_record_to_dict(record) async for record in db.select_and_fetchall(sql, sql_args)]
+
+    if len(job_groups) == 51:
+        job_groups.pop()
+        last_child_job_group_id = job_groups[-1]['job_group_id']
+    else:
+        last_child_job_group_id = None
+
+    return (job_groups, last_child_job_group_id)
+
+
+@routes.get('/api/v1alpha/batches/{batch_id}/job-groups/{job_group_id}/job-groups')
```

Comment:
It seems like there's an asymmetry between `get_job_groups_v1` and `get_job_group_jobs_v1`. The former has no corresponding batch method whereas the latter does.

Of course, the latter is for backwards compatibility. What do you think? I think I'm mildly inclined to support both backwards compatibility *and* symmetry ergo we should have `/api/v1alpha/batches/{batch_id}/job-groups`.

Reply:
Is this the same as `recursive = True`? I don't quite follow this comment.

Reply:
I can ask a job group for its "immediate children" jobs:
```
/api/v1alpha/batches/{batch_id}/job-groups/{job_group_id}/jobs
```
I can ask a batch for its "immediate children" jobs:
```
/api/v1alpha/batches/{batch_id}/jobs
```
I can ask a job group for its "immediate children" job groups:
```
/api/v1alpha/batches/{batch_id}/job-groups/{job_group_id}/job-groups
```
I can ... whoops!  I can't ask a batch for its "immediate children" job groups.

I'm saying we should add:

```
/api/v1alpha/batches/{batch_id}/job-groups
```